lib/spack/spack/build_environment.py

```diff
@@ -109,7 +109,8 @@
 
 
 # Platform-specific library suffix.
-dso_suffix = "dylib" if sys.platform == "darwin" else "so"
+dso_suffix = "dylib" if sys.platform == "darwin" else "dll" if sys.platform == "win32" else "so"
```

Comment:
Could this be expanded into an if/else:

```
if sys.platform == "darwin":
  dso_suffix =
elif sys.platform == "win32":
  ...
else:
  dso_suffix = ".so"
```

Reply:
Amusingly enough, this change is actually from the RPath analog PR, not sure why it's flagged as a change here... Either way, will amend.